File: src/main/java/com/ar/genesis/sistema/persistence/repositoryImplementacion/ClienteRepoImplementacion.java
package com.ar.genesis.sistema.persistence.repositoryImplementacion;

import com.ar.genesis.sistema.core.domain.Cliente;
import com.ar.genesis.sistema.core.repository.IClienteRepository;
import com.ar.genesis.sistema.persistence.repositoryCRUD.IClienteCRUD;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ClienteRepoImplementacion implements IClienteRepository {

    @Autowired
    IClienteCRUD clienteCRUD;

    @Override
    public boolean existeCliente(String nombre) {
        return clienteCRUD.existsByNombre(nombre);
    }

    @Override
    public boolean guardarCliente(Cliente unCliente) {
        return clienteCRUD.save(unCliente).getId()!=null;
    }
}

File: sistema/src/main/kotlin/com/ar/genesis/sistema/repository/impl/ClienteRepoImplementacion.kt
package com.ar.genesis.sistema.repository.impl

import com.ar.genesis.sistema.domain.Cliente
import com.ar.genesis.sistema.repository.ICliente

class ClienteRepoImplementacion  (val repo : ClienteRepository) : ICliente
{

    override fun existeCliente(nombre: String): Boolean {
        var encontrado = false
        val clienteEncontrado : Cliente = repo.findByNombre(nombre)
        if (clienteEncontrado.nombre == nombre)
            encontrado = true
        return encontrado
    }

    override fun guardarCliente(cliente: Cliente): Boolean {
        repo.save(cliente)
        return true
    }

}